# Minor Project - Exploratory Data Analysis on Movies dataset


Project includes: **Data cleaning || EDA || Answers to some questions (below)||[Summary](#6)** <p>

Questions inculude:
>[1) Which are the movies with the third-lowest and third-highest budget?](#1) <br>
[2) What is the average number of words in movie titles between the years 2000-2005?](#2)<br>
[3) What is the most common Genre for Vin Diesel and Emma Watson movies?](#3) <br>
[4) Which are the movies with the most and least earned revenue?](#4) <br>
[5) What is the average runtime of movies in the year 2006?](#5) <br>

Description of the dataset is available at https://www.kaggle.com/tmdb/tmdb-movie-metadata


In [1]:
#Importing libraries
import pandas as pd
import numpy as np
#Read csv file
df=pd.read_csv("minorproject.csv", index_col=0)
df.head() #First 5 rows

imdb_id  popularity     budget     revenue  \
id                                                     
135397  tt0369610   32.985763  150000000  1513528810   
76341   tt1392190   28.419936  150000000   378436354   
262500  tt2908446   13.112507  110000000   295238201   
140607  tt2488496   11.173104  200000000  2068178225   
168259  tt2820852    9.335014  190000000  1506249360   

                      original_title  \
id                                     
135397                Jurassic World   
76341             Mad Max: Fury Road   
262500                     Insurgent   
140607  Star Wars: The Force Awakens   
168259                     Furious 7   

                                                     cast  \
id                                                          
135397  Chris Pratt|Bryce Dallas Howard|Irrfan Khan|Vi...   
76341   Tom Hardy|Charlize Theron|Hugh Keays-Byrne|Nic...   
262500  Shailene Woodley|Theo James|Kate Winslet|Ansel...   
140607  Harrison Ford|Mark Hamill|Carrie Fisher|Adam D...   
168259  Vin Diesel|Paul Walker|Jason Statham|Michelle ...   

                                                 homepage          director  \
id                                                                            
135397                      http://www.jurassicworld.com/   Colin Trevorrow   
76341                         http://www.madmaxmovie.com/     George Miller   
262500     http://www.thedivergentseries.movie/#insurgent  Robert Schwentke   
140607  http://www.starwars.com/films/star-wars-episod...       J.J. Abrams   
168259                           http://www.furious7.com/         James Wan   

                              tagline  \
id                                      
135397              The park is open.   
76341              What a Lovely Day.   
262500     One Choice Can Destroy You   
140607  Every generation has a story.   
168259            Vengeance Hits Home   

                                                 keywords  \
id                                                          
135397  monster|dna|tyrannosaurus rex|velociraptor|island   
76341    future|chase|post-apocalyptic|dystopia|australia   
262500  based on novel|revolution|dystopia|sequel|dyst...   
140607              android|spaceship|jedi|space opera|3d   
168259                car race|speed|revenge|suspense|car   

                                                 overview  runtime  \
id                                                                   
135397  Twenty-two years after the events of Jurassic ...      124   
76341   An apocalyptic story set in the furthest reach...      120   
262500  Beatrice Prior must confront her inner demons ...      119   
140607  Thirty years after defeating the Galactic Empi...      136   
168259  Deckard Shaw seeks revenge against Dominic Tor...      137   

                                           genres  \
id                                                  
135397  Action|Adventure|Science Fiction|Thriller   
76341   Action|Adventure|Science Fiction|Thriller   
262500         Adventure|Science Fiction|Thriller   
140607   Action|Adventure|Science Fiction|Fantasy   
168259                      Action|Crime|Thriller   

                                     production_companies release_date  \
id                                                                       
135397  Universal Studios|Amblin Entertainment|Legenda...       6/9/15   
76341   Village Roadshow Pictures|Kennedy Miller Produ...      5/13/15   
262500  Summit Entertainment|Mandeville Films|Red Wago...      3/18/15   
140607          Lucasfilm|Truenorth Productions|Bad Robot     12/15/15   
168259  Universal Pictures|Original Film|Media Rights ...       4/1/15   

        vote_count  vote_average  release_year    budget_adj   revenue_adj  
id                                                                          
135397        5562           6.5          2015  1.379999e+08  1.392446e+09  
76341         6185       

In [2]:
df.shape

(10866, 20)

**Cleaning (fix, replace, drop, etc) of values <br>
  Drop unused columns, Replace Values & Using mean/median of datas to replace empty (0) entries**

In [3]:
#Dropping columns
df=df.drop(["imdb_id","popularity","homepage","director","tagline","keywords","overview","production_companies","vote_count","vote_average","release_date"
],axis=1)
#Replace 0 to NaN
df.replace(0,np.nan, inplace=True) 

#Fixing missing values (of columns of interest) to mean of data
df["budget"]=df["budget"].fillna(np.mean(df["budget"]))
df["revenue"]=df["revenue"].fillna(np.mean(df["revenue"]))

df.head()


budget       revenue                original_title  \
id                                                                
135397  150000000.0  1.513529e+09                Jurassic World   
76341   150000000.0  3.784364e+08            Mad Max: Fury Road   
262500  110000000.0  2.952382e+08                     Insurgent   
140607  200000000.0  2.068178e+09  Star Wars: The Force Awakens   
168259  190000000.0  1.506249e+09                     Furious 7   

                                                     cast  runtime  \
id                                                                   
135397  Chris Pratt|Bryce Dallas Howard|Irrfan Khan|Vi...    124.0   
76341   Tom Hardy|Charlize Theron|Hugh Keays-Byrne|Nic...    120.0   
262500  Shailene Woodley|Theo James|Kate Winslet|Ansel...    119.0   
140607  Harrison Ford|Mark Hamill|Carrie Fisher|Adam D...    136.0   
168259  Vin Diesel|Paul Walker|Jason Statham|Michelle ...    137.0   

                                           genres  release_year    budget_adj  \
id                                                                              
135397  Action|Adventure|Science Fiction|Thriller          2015  1.379999e+08   
76341   Action|Adventure|Science Fiction|Thriller          2015  1.379999e+08   
262500         Adventure|Science Fiction|Thriller          2015  1.012000e+08   
140607   Action|Adventure|Science Fiction|Fantasy          2015  1.839999e+08   
168259                      Action|Crime|Thriller          2015  1.747999e+08   

         revenue_adj  
id                    
135397  1.392446e+09  
76341   3.481613e+08  
262500  2.716190e+08  
140607  1.902723e+09  
168259  1.385749e+09

<a id='1'></a>
# 1) Which are the movies with the third-lowest and third-highest budget?

In [4]:

sort1=df.sort_values("budget", ascending=True)
sort1.head(3)
#Third lowest budget=1.0	; Movie name Love, Wedding, Marriage

budget       revenue                      original_title  \
id                                                                 
287524     1.0  8.922066e+07                         Fear Clinic   
12689      1.0  8.922066e+07  Angus, Thongs and Perfect Snogging   
59296      1.0  1.378000e+03             Love, Wedding, Marriage   

                                                     cast  runtime  \
id                                                                   
287524  Thomas Dekker|Robert Englund|Cleopatra Coleman...     95.0   
12689   Georgia Groome|Eleanor Tomlinson|Aaron Taylor-...    100.0   
59296   Mandy Moore|Kellan Lutz|Jessica Szohr|Autumn F...     90.0   

                             genres  release_year  budget_adj  revenue_adj  
id                                                                          
287524                       Horror          2014    0.921091          NaN  
12689   Comedy|Drama|Family|Romance          2008    1.012787          NaN  
59296                Comedy|Romance          2011    0.969398  1335.830503

In [5]:
sort2=df.sort_values("budget", ascending=False)
sort2.head(3)
#Third highest budget 300000000; Movie name=Pirates of the Caribbean: At World's End

budget       revenue                               original_title  \
id                                                                              
46528  425000000.0  1.108757e+07                            The Warrior's Way   
1865   380000000.0  1.021683e+09  Pirates of the Caribbean: On Stranger Tides   
285    300000000.0  9.610000e+08     Pirates of the Caribbean: At World's End   

                                                    cast  runtime  \
id                                                                  
46528  Kate Bosworth|Jang Dong-gun|Geoffrey Rush|Dann...    100.0   
1865   Johnny Depp|Penélope Cruz|Geoffrey Rush|Ian M...    136.0   
285    Johnny Depp|Orlando Bloom|Keira Knightley|Geof...    169.0   

                                          genres  release_year    budget_adj  \
id                                                                             
46528  Adventure|Fantasy|Action|Western|Thriller          2010  4.250000e+08   
1865                    Adventure|Action|Fantasy          2011  3.683713e+08   
285                     Adventure|Fantasy|Action          2007  3.155006e+08   

        revenue_adj  
id                   
46528  1.108757e+07  
1865   9.904175e+08  
285    1.010654e+09

**Movie with third-lowest budget =Love, Wedding, Marriage** <p>
**Movie with third-highest budget =Pirates of the Caribbean: At World's End**

<a id='2'></a>
# 2) What is the average number of words in movie titles between the years 2000-2005?

In [6]:
df=pd.read_csv("minorproject.csv", index_col=18) #here index col is year of release
sort3=df.loc[[2000,2001,2002,2003,2004,2005],["original_title"]] #rows of interest and column of interest
count=0 #initialise count
sort3 #show df
#Here there are 1678 rows ie 1678 is the total number of movies

original_title
release_year                                
2000                               Gladiator
2000                                 Memento
2000          How the Grinch Stole Christmas
2000                               Cast Away
2000                                  Snatch
...                                      ...
2005           The Devil and Daniel Johnston
2005                          Wassup Rockers
2005              Mon petit doigt m'a dit...
2005                         Quatre étoiles
2005                                  Paheli

[1687 rows x 1 columns]

In [7]:
for item in sort3.original_title:
    count=count+len((item.split())) #split wherever whitespace, add item to count
 #Here, Count is the Total number of words
print("Total number of words in movie titles in the given years=", count)
print("The average number of words in movie titles between the years 2000 to 2005=",count/1687, "words.")

Total number of words in movie titles in the given years= 4858
The average number of words in movie titles between the years 2000 to 2005= 2.879668049792531 words.


<a id='3'></a>
# 3) What is the most common Genre for Vin Diesel & Emma Watson movies?

**3.a) Vin Diesel**

In [8]:
df=pd.read_csv("minorproject.csv", index_col=13) #index col is genre
df.fillna(value="False", inplace=True) #fill blanks with False
df1 = df['cast'].str.contains("Vin Diesel") #True wherever cast contains Vin Diesel

df1=df1.to_frame() #to convert series to df
df1=df1[df1["cast"]==True] #selecting rows(genres) where cast (Vin Diesel) is there (True)

df1.groupby("genres")["cast"].sum() #groups by "genre", takes sum of unique genres in df1 (where df1 contains genres where Vin Diesel is present)

genres
Action                                                2.0
Action|Adventure|Science Fiction|Thriller             1.0
Action|Animation|Science Fiction                      1.0
Action|Animation|Science Fiction|Thriller             1.0
Action|Comedy|Drama|Family|Thriller                   1.0
Action|Crime|Drama|Thriller                           1.0
Action|Crime|Thriller                                 2.0
Action|Drama                                          2.0
Action|Science Fiction|Adventure                      1.0
Action|Thriller|Crime                                 2.0
Adventure|Action|Thriller|Science Fiction             1.0
Adventure|Animation|Family|Fantasy|Science Fiction    1.0
Crime|Drama|Thriller                                  1.0
Crime|Thriller                                        1.0
Drama|Action|Comedy|Crime                             1.0
Fantasy|Action|Adventure                              1.0
Science Fiction|Action|Thriller                       1.0
Thrille

**Most Common Genre for Vin Diesel= Action, Action|Crime|Thriller, Action|Drama,Action|Thriller|Crime**
                         

**3b) Emma Watson**

In [9]:
#Similarly for Emma Watson
df=pd.read_csv("minorproject.csv", index_col=13)
df.fillna(value="False", inplace=True)
df1 = df['cast'].str.contains("Emma Watson")

df1=df1.to_frame() #to convert series to df
df1=df1[df1["cast"]==True] #selecting rows(genres) where cast (Emma Watson) is there (True)

df1.groupby("genres")["cast"].sum()

genres
Adventure|Family|Fantasy            1.0
Adventure|Fantasy|Family            6.0
Adventure|Fantasy|Family|Mystery    1.0
Animation|Comedy|Family             1.0
Drama|Adventure                     1.0
Drama|Crime                         1.0
Drama|Family                        1.0
Drama|Romance                       1.0
Horror|Mystery|Thriller             1.0
Name: cast, dtype: float64

**Most Common Genre for Emma Watson -  Adventure|Fantasy|Family(6)**

<a id='4'></a>

# 4) Which are the movies with the most and least earned revenue?

In [10]:
df=pd.read_csv("minorproject.csv", index_col=0) #Read CSV
df.replace(0,np.nan, inplace=True) #Replace 0 values with na

df["revenue"]=df["revenue"].fillna(np.mean(df["revenue"])) #Replace na values with mean of Revenue
sort4=df.sort_values("revenue", ascending=True) #Sort in ascending to find least earned revenue
sort4.head(1)

imdb_id  popularity     budget  revenue   original_title  \
id                                                                  
13537  tt0323944    0.462609  6000000.0      2.0  Shattered Glass   

                                                    cast homepage   director  \
id                                                                             
13537  Hayden Christensen|Peter Sarsgaard|Chloë Sevi...      NaN  Billy Ray   

      tagline keywords                                           overview  \
id                                                                          
13537     NaN      NaN  Fact-based 2003 drama about the young journali...   

       runtime         genres  \
id                              
13537     94.0  Drama|History   

                                    production_companies release_date  \
id                                                                      
13537  Lions Gate Films|Cruise/Wagner Productions|Bau...     11/14/03   

       vote_count  vote_average  release_year    budget_adj  revenue_adj  
id                                                                        
13537          46           6.4          2003  7.112116e+06     2.370705

In [11]:
sort5=df.sort_values("revenue", ascending=False)
sort5.head(1)

imdb_id  popularity       budget       revenue original_title  \
id                                                                       
19995  tt0499549    9.432768  237000000.0  2.781506e+09         Avatar   

                                                    cast  \
id                                                         
19995  Sam Worthington|Zoe Saldana|Sigourney Weaver|S...   

                          homepage       director  \
id                                                  
19995  http://www.avatarmovie.com/  James Cameron   

                           tagline  \
id                                   
19995  Enter the World of Pandora.   

                                                keywords  \
id                                                         
19995  culture clash|future|space war|space colony|so...   

                                                overview  runtime  \
id                                                                  
19995  In the 22nd century, a paraplegic Marine is di...    162.0   

                                         genres  \
id                                                
19995  Action|Adventure|Fantasy|Science Fiction   

                                    production_companies release_date  \
id                                                                      
19995  Ingenious Film Partners|Twentieth Century Fox ...     12/10/09   

       vote_count  vote_average  release_year    budget_adj   revenue_adj  
id                                                                         
19995        8458           7.1          2009  2.408869e+08  2.827124e+09

**Movie with most earned revenue= Avatar** <p>	
**Movie with least earned revenue= Shattered Glass**

<a id='5'></a>
# 5) What is the average runtime of movies in the year 2006?


In [12]:
df=pd.read_csv("minorproject.csv", index_col=-3) #index is the release year
rt=df.loc[2006, "runtime"] #Selecting row 2006 and column of interest (runtime)
rt=rt.to_frame() #Series to DF
rt["runtime"].mean()

101.68382352941177

**The average runtime of movies in the year 2006=101.68 minutes**

<a id='6'></a>

# Summary
1) Which are the movies with the third-lowest and third-highest budget?<br>
>**Movie with third-lowest budget =Love, Wedding, Marriage** <br>
>**Movie with third-highest budget =Pirates of the Caribbean: At World's End** <br>

2) What is the average number of words in movie titles between the years 2000-2005?<br>
>**The average number of words in movie titles between the years 2000 to 2005= 2.879668049792531 words.**<br>

3) What is the most common Genre for Vin Diesel and Emma Watson movies? <br>
>**Most Common Genre for Vin Diesel= Action, Action|Crime|Thriller, Action|Drama,Action|Thriller|Crime**<br>
>**Most Common Genre for Emma Watson -  Adventure|Fantasy|Family(6)** <br>

4) Which are the movies with the most and least earned revenue? <br>
>**Movie with most earned revenue= Avatar** <br>	
>**Movie with least earned revenue= Shattered Glass**<br>

5) What is the average runtime of movies in the year 2006? <br>
>**101.68382352941177 minutes** <br>
